github/Issue.py

```diff
@@ -347,119 +316,100 @@ def create_comment(self, body):
             self._requester, headers, data, completed=True
         )
 
-    def delete_labels(self):
+    def delete_labels(self) -> None:
         """
         :calls: `DELETE /repos/{owner}/{repo}/issues/{number}/labels <https://docs.github.com/en/rest/reference/issues#labels>`_
-        :rtype: None
         """
         headers, data = self._requester.requestJsonAndCheck(
             "DELETE", f"{self.url}/labels"
         )
 
     def edit(
         self,
-        title=github.GithubObject.NotSet,
-        body=github.GithubObject.NotSet,
-        assignee=github.GithubObject.NotSet,
-        state=github.GithubObject.NotSet,
-        milestone=github.GithubObject.NotSet,
-        labels=github.GithubObject.NotSet,
-        assignees=github.GithubObject.NotSet,
-        state_reason=github.GithubObject.NotSet,
+        title: Opt[str] = NotSet,
+        body: Opt[str] = NotSet,
+        assignee: Opt[str | NamedUser] = NotSet,
+        state: Opt[str] = NotSet,
+        milestone: Opt[Milestone] = NotSet,
+        labels: Opt[list[str]] = NotSet,
+        assignees: Opt[list[str]] = NotSet,
+        state_reason: Opt[str] = NotSet,
     ):
         """
         :calls: `PATCH /repos/{owner}/{repo}/issues/{number} <https://docs.github.com/en/rest/reference/issues>`_
-        :param title: string
-        :param body: string
-        :param assignee: string or :class:`github.NamedUser.NamedUser` or None
-        :param state: string
-        :param milestone: :class:`github.Milestone.Milestone` or None
-        :param labels: list of string
-        :param assignees: list of string or :class:`github.NamedUser.NamedUser`
-        :param state_reason: string
-        :rtype: None
-        """
-        assert title is github.GithubObject.NotSet or isinstance(title, str), title
-        assert body is github.GithubObject.NotSet or isinstance(body, str), body
-        assert (
-            assignee is github.GithubObject.NotSet
-            or assignee is None
-            or isinstance(assignee, github.NamedUser.NamedUser)
-            or isinstance(assignee, str)
+        """
+        assert isinstance(title, (str, _NotSetType)), title
+        assert isinstance(body, (str, _NotSetType)), body
+        assert assignee is None or isinstance(
+            assignee, (github.NamedUser.NamedUser, str, _NotSetType)
         ), assignee
```

Comment:
Why should `None` be allowed here?
```suggestion
        assert isinstance(assignee, (github.NamedUser.NamedUser, str, _NotSetType)), assignee
```

Reply:
I have no idea, it's the old behavoir